libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -27,6 +27,7 @@
 #include <AP_BoardConfig/AP_BoardConfig.h>
 
 #include "AP_Baro_BMP085.h"
+#include "AP_Baro_BMP280.h"
```

Comment:
The data handling is to different. BMP085 need to request temp, wait, read temp, calculate temp, request pressure, wait, read pressure, calculate pressure. That is the reason for the state machine. The new BMP280 does continues measurement,  you only have to read (both temp and pressure at the same time) and calculate. That the reason I choose a new file.